From 53cb93730a495307edcb86cb52b70ed1cd9058a0 Mon Sep 17 00:00:00 2001
From: Yu Watanabe <watanabe.yu+github@gmail.com>
Date: Sat, 20 Oct 2018 04:24:04 +0900
Subject: [PATCH 4/7] Build: do not build cray related modules

---
 contribs/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/contribs/Makefile.am b/contribs/Makefile.am
index 7fa8d82436..d5f7218791 100644
--- a/contribs/Makefile.am
+++ b/contribs/Makefile.am
@@ -1,4 +1,4 @@
-SUBDIRS = cray lua mic openlava pam pam_slurm_adopt perlapi pmi pmi2 seff sgather sgi sjobexit torque
+SUBDIRS = lua mic openlava pam pam_slurm_adopt perlapi pmi pmi2 seff sgather sgi sjobexit torque
 
 EXTRA_DIST = \
 	gcp			\
-- 
2.17.2

