--- slurm-17.02.7/src/plugins/node_features/knl_generic/node_features_knl_generic.c	2017-08-15 14:37:07.000000000 -0400
+++ slurm-17.02.7/src/plugins/node_features/knl_generic/node_features_knl_generic.c.new	2017-08-21 01:35:38.352301060 -0400
@@ -1607,7 +1607,7 @@
 // insmod /home/tim/kmod/xppsl-addons/zonesort_module.ko
 		struct stat sb;
 		int buf_len, fd, i, len;
-		char buf[8];
+		char buf[16];
 
 		if (stat(ZONE_SORT_PATH, &sb) == -1)
 			(void) system(MODPROBE_PATH " zonesort_module");
--- slurm-17.02.7/src/plugins/select/bluegene/bg_core.c	2017-07-21 11:17:53.000000000 -0400
+++ slurm-17.02.7/src/plugins/select/bluegene/bg_core.c.new	2017-08-21 01:39:35.896506235 -0400
@@ -701,7 +701,7 @@
  */
 extern const char *bg_err_str(int inx)
 {
-	static char tmp_char[10];
+	static char tmp_char[32];
 
 	switch (inx) {
 	case SLURM_SUCCESS:
--- slurm-17.02.7/src/plugins/task/affinity/cpuset.c	2017-07-21 11:17:53.000000000 -0400
+++ slurm-17.02.7/src/plugins/task/affinity/cpuset.c.new	2017-08-21 01:42:53.872712983 -0400
@@ -354,7 +354,7 @@
 int	slurm_set_memset(char *path, nodemask_t *new_mask)
 {
 	char file_path[PATH_MAX];
-	char mstr[1 + CPU_SETSIZE * 4], tmp[10];
+	char mstr[1 + CPU_SETSIZE * 4], tmp[16];
 	int fd, i, max_node;
 	ssize_t rc;

--- slurm-17.02.7/src/slurmd/slurmd/read_proc.c	2017-07-21 11:17:53.000000000 -0400
+++ slurm-17.02.7/src/slurmd/slurmd/read_proc.c.new	2017-08-26 00:27:38.036289952 -0400
@@ -233,7 +233,7 @@
 	DIR *proc_fs;
 	struct dirent *proc_ent;
 	int proc_fd, proc_stat_size, found, n;
-	char proc_name[22], *proc_stat;
+	char proc_name[272], *proc_stat;
 	struct stat buffer;
 	int uid, session;
 	long resident_set_size;
