diff --git a/etc/slurmctld.service.in b/etc/slurmctld.service.in
index ebf92b0da4..d13a653339 100644
--- a/etc/slurmctld.service.in
+++ b/etc/slurmctld.service.in
@@ -8,7 +8,7 @@ Type=forking
 EnvironmentFile=-/etc/sysconfig/slurmctld
 ExecStart=@sbindir@/slurmctld $SLURMCTLD_OPTIONS
 ExecReload=/bin/kill -HUP $MAINPID
-PIDFile=/var/run/slurmctld.pid
+PIDFile=/var/run/slurm/slurmctld.pid
 @SYSTEMD_TASKSMAX_OPTION@
 
 [Install]
diff --git a/etc/slurmd.service.in b/etc/slurmd.service.in
index 2e20bb33a3..263bd8aa77 100644
--- a/etc/slurmd.service.in
+++ b/etc/slurmd.service.in
@@ -8,7 +8,7 @@ Type=forking
 EnvironmentFile=-/etc/sysconfig/slurmd
 ExecStart=@sbindir@/slurmd $SLURMD_OPTIONS
 ExecReload=/bin/kill -HUP $MAINPID
-PIDFile=/var/run/slurmd.pid
+PIDFile=/var/run/slurm/slurmd.pid
 KillMode=process
 LimitNOFILE=51200
 LimitMEMLOCK=infinity
diff --git a/etc/slurmdbd.service.in b/etc/slurmdbd.service.in
index 5f8f73c7f1..a0aede6feb 100644
--- a/etc/slurmdbd.service.in
+++ b/etc/slurmdbd.service.in
@@ -8,7 +8,7 @@ Type=forking
 EnvironmentFile=-/etc/sysconfig/slurmdbd
 ExecStart=@sbindir@/slurmdbd $SLURMDBD_OPTIONS
 ExecReload=/bin/kill -HUP $MAINPID
-PIDFile=/var/run/slurmdbd.pid
+PIDFile=/var/run/slurm/slurmdbd.pid
 @SYSTEMD_TASKSMAX_OPTION@
 
 [Install]
