diff --git a/src/plugins/node_features/knl_generic/Makefile.am b/src/plugins/node_features/knl_generic/Makefile.am
index 11da24c075..4633e2de46 100644
--- a/src/plugins/node_features/knl_generic/Makefile.am
+++ b/src/plugins/node_features/knl_generic/Makefile.am
@@ -8,7 +8,7 @@ AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/src/common
 
 pkglib_LTLIBRARIES = node_features_knl_generic.la
 node_features_knl_generic_la_SOURCES = node_features_knl_generic.c
-node_features_knl_generic_la_LDFLAGS = $(SO_LDFLAGS) $(PLUGIN_FLAGS)
+node_features_knl_generic_la_LDFLAGS = $(SO_LDFLAGS) $(NUMA_LIBS) $(PLUGIN_FLAGS)
 
 force:
 $(node_features_knl_generic_la_LIBADD) : force
